ngx_http_auth_ldap_module.c

```diff
@@ -1334,18 +1345,31 @@ ngx_http_auth_ldap_ssl_handshake_handler(ngx_connection_t *conn, ngx_flag_t vali
           long chain_verified = SSL_get_verify_result(conn->ssl->connection);
 
           int addr_verified;
-          char *hostname = c->server->ludpp->lud_host;
-          addr_verified = X509_check_host(cert, hostname, 0, 0, 0);
-
-          if (!addr_verified) { // domain not in cert? try IP
-            size_t len; // get IP length
-            if (conn->sockaddr->sa_family == 4) len = 4;
-            else if (conn->sockaddr->sa_family == 6) len = 16;
-            else { // very unlikely indeed
-              ngx_http_auth_ldap_close_connection(c);
-              return;
+          if (c->server->ssl_check_cert == SSL_CERT_VERIFY_CHAIN) {
+            // chain_verified is enough, not requiring full name/IP verification
+            addr_verified = 1;
+
+          } else {
+            // verify hostname/IP
+            char *hostname = c->server->ludpp->lud_host;
+            addr_verified = X509_check_host(cert, hostname, 0, 0, 0);
+
+            if (!addr_verified) { // domain not in cert? try IP
+              size_t len; // get IP length
+
+              struct sockaddr *conn_sockaddr = NULL;
+              if (conn->sockaddr != NULL) conn_sockaddr = conn->sockaddr;
+              else if (c->conn.sockaddr != NULL) conn_sockaddr = c->conn.sockaddr;
+              else conn_sockaddr = &c->server->parsed_url.sockaddr.sockaddr;
+
+              if (conn_sockaddr->sa_family == AF_INET) len = 4;
+              else if (conn_sockaddr->sa_family == AF_INET6) len = 16;
+              else { // very unlikely indeed
```

Comment:
We are now checking `AF_INET` and `AF_INET6` rather than just `4` and `6`, which wouldn't have ever been correct anyway.